ui/src/components/MobileHeader.tsx

```diff
@@ -0,0 +1,44 @@
+import { Link } from 'react-router-dom';
+import CaretLeftIcon from './icons/CaretLeftIcon';
+
+export default function MobileHeader({
+  title,
+  pathBack,
+  pathBackText,
+  action,
+  secondaryAction,
+}: {
+  title: string;
+  pathBack?: string;
+  pathBackText?: string;
+  action?: React.ReactNode;
+  secondaryAction?: React.ReactNode;
+}) {
+  return (
+    <div className="grid w-full grid-cols-3 items-center justify-between bg-white py-3 pl-2 pr-4 font-system-sans">
+      {pathBack ? (
+        <Link className="flex items-center" to={pathBack}>
+          <CaretLeftIcon className="h-6 w-6 text-gray-400" />
+          {pathBackText && (
+            <span className="text-[17px] leading-6 text-gray-800 line-clamp-1">
+              Cancel
+            </span>
+          )}
```

Comment:
Maybe we could just conditionally apply `sr-only` here if they don't want the text that way it always has an accessible label?